File: client/src/app/_models/Question.ts
export interface Question {
  id: number;
  caption: string;
  askedBy: string;
  createdAt: string;
  answers: Answer[];
}
  
  export interface Answer {
    id: number;
    answerText: string;
    answeredBy: string;
    createdAt: string;
  }
  

File: API/Entities/Question.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace API.Entities;

public class Question
{
    [Required]
    public int Id { get; set; }
    [Required]
    public string? Caption { get; set; }
    [Required]
    public string? AskedBy { get; set; }  // Username
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public List<Answer> Answers { get; set; } = new();
}
